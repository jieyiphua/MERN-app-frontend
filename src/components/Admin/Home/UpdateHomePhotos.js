import React from 'react';
import { Redirect, withRouter, Link } from 'react-router-dom';
import axios from 'axios';

class UpdateHomePhotos extends React.Component {
  state = {
    data: [],
    image: this.props.selectedContent.image || '',
    id: this.props.selectedContent._id,
    errors: []
  }

  // async componentDidMount() {
  //   const response = await axios.get(
  //     'https://deployment-mern-backend-tessivanjayz.gardtess.now.sh/carousel-images'
  //   );
  //   console.log(this.props.history)

  //   this.setState({
  //     data: response.data
  //   });
  // }
  componentDidMount() {
    console.log('Update Home Photos did mount')
    console.log(this.props)
  }

  onInputChange = event => {
    this.setState({ image: event.target.value });
    console.log(this.state);
  };

  onFormSubmit = async event => {
    event.preventDefault();
    console.log(this.state)
    console.log(this.props)
    try {
      const response = await axios.put(`https://deployment-mern-backend-tessivanjayz.gardtess.now.sh/carousel-images/${this.state.id}`, { image: this.state.image} )
      console.log(response)
    } catch (error) {
      console.log(error)
      this.setState({
        errors: error.response
      })
    }
    this.props.history.push('/');
  }

  render() {
    const { authentication } = this.props;

    if (!authentication) {
      return <Redirect to="/admin/login" />;
    }
    console.log(this.state)
    console.log(this.props)
    return(
      <div className="main-container" id="new-class-main-container">
        <div className="new-class-header">
          <h1>Update Carousel Image</h1>
        </div>
        <div className="new-class-form-container">
          <form  onSubmit={this.onFormSubmit} className="new-class-form">
            <label htmlFor="description"> Image Url:</label>
            <input type="text" value={this.state.image} onChange={this.onInputChange} name="image" id="image"/>
            <div className="classes-buttons">
              <button onClick={this.onFormSubmit}>Submit</button>
              <Link to="/"><button>Back</button></Link>
            </div>
          </form>
        </div>
      </div>
    )
  }
}

export default withRouter(UpdateHomePhotos);