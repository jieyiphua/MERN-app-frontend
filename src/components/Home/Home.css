#punchLine {
  font-family: 'Merriweather', serif;
  font-size: 2.4vw;
  margin-top: 20px;
}

#admin-link {
  color: rgb(248, 121, 121);
  font-size: 70%;
}

.landing-main-container {
  width: 100%;
  margin-top: 150px;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.hero-content-home {
  width: 90%;
  margin-top: 20px;
  display: flex;
  align-items: center;
  flex-direction: column;
}

.image-gallery-home {
  width: 60%;
  height: 420px;
  background-image: url("./home1.JPG");
  background-repeat: no-repeat;
  background-size: 100%;
}

.footerContainer {
  margin-top: 180px;
  margin-bottom: 30px; 
}