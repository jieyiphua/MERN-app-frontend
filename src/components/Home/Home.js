import React from "react";
import "./Home.css";
import { Link } from "react-router-dom";
import axios from "axios";

import { Carousel } from "react-bootstrap";

class Home extends React.Component {
  state = {
    data: []
  };

  async componentDidMount() {
    const response = await axios.get(
      "https://deployment-mern-backend-tessivanjayz.gardtess.now.sh/carousel-images"
    );
    console.log(response);
    console.log(response.data[0].image);

    this.setState({
      data: response.data
    });
    this.props.handleLoading();
  }

  render() {
    const { authentication, onContentSelect } = this.props;
    console.log(this.state.data[0]);

    if (!authentication) {
      return (
        <div className="landing-main-container">
          <div className="alert-box">
            <div className="free-trial">
              <Link to="/visit/contact">
                We offer free trials for your first group class, contact us
                today to get started!
              </Link>
            </div>
          </div>
          <div className="hero-content-home">
            <div>
              <Carousel>
                <Carousel.Item>
                  <img
                    className="d-block w-100"
                    src={
                      this.state.data.length === 3 && this.state.data[0].image
                    }
                    alt="First slide"
                  />
                  <Carousel.Caption />
                </Carousel.Item>
                <Carousel.Item>
                  <img
                    className="d-block w-100"
                    src={
                      this.state.data.length === 3 && this.state.data[1].image
                    }
                    alt="Third slide"
                  />
                  <Carousel.Caption />
                </Carousel.Item>
                <Carousel.Item>
                  <img
                    className="d-block w-100"
                    src={
                      this.state.data.length === 3 && this.state.data[2].image
                    }
                    alt="Third slide"
                  />
                  <Carousel.Caption />
                </Carousel.Item>
              </Carousel>
            </div>
            <h3 id="punch-line">Professional Exciting Positive Friendly Fun</h3>
          </div>
          <div className="footer-container">
            <div className="footerBox">
              <Link to="/admin/login/" className="admin-link">
                <p id="admin-link">admin login ></p>
              </Link>
            </div>
          </div>
        </div>
      );
    } else {
      return (
        <div className="landing-main-container">
          <div className="alert-box">
            <div className="free-trial">
              <Link to="/visit/contact">
                We offer free trials for your first group class, contact us
                today to get started!
              </Link>
            </div>
          </div>
          <div className="hero-content-home">
            <div>
              <Carousel>
                <Carousel.Item>
                  <img
                    className="d-block w-100"
                    src={
                      this.state.data.length === 3 && this.state.data[0].image
                    }
                    alt="First slide"
                  />
                  <Carousel.Caption />
                </Carousel.Item>
                <Carousel.Item>
                  <img
                    className="d-block w-100"
                    src={
                      this.state.data.length === 3 && this.state.data[1].image
                    }
                    alt="Third slide"
                  />
                  <Carousel.Caption />
                </Carousel.Item>
                <Carousel.Item>
                  <img
                    className="d-block w-100"
                    src={
                      this.state.data.length === 3 && this.state.data[2].image
                    }
                    alt="Third slide"
                  />
                  <Carousel.Caption />
                </Carousel.Item>
              </Carousel>
            </div>
            <h3 id="punch-line">Professional Exciting Positive Friendly Fun</h3>
          </div>
          <div>
            <Link
              to="/admin/home/update-photos"
              onClick={() => onContentSelect(this.state.data[0])}
            >
              <button className="plain-button">Update Image 1</button>
            </Link>
            <Link
              to="/admin/home/update-photos"
              onClick={() => onContentSelect(this.state.data[1])}
            >
              <button className="plain-button">Update Image 2</button>
            </Link>
            <Link
              to="/admin/home/update-photos"
              onClick={() => onContentSelect(this.state.data[2])}
            >
              <button className="plain-button">Update Image 3</button>
            </Link>
          </div>
        </div>
      );
    }
  }
}

export default Home;
