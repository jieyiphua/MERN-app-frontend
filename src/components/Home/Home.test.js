import Home from "./Home";
import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";

describe("<Home />", () => {
  it("renders component", () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
});
