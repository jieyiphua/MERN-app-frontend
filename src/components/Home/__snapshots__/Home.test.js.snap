// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders component 1`] = `
<div
  className="landing-main-container"
>
  <div
    className="alert-box"
  >
    <div
      className="free-trial"
    >
      <Link
        to="/visit/contact"
      >
        We offer free trials for your first group class, contact us today to get started!
      </Link>
    </div>
  </div>
  <div
    className="hero-content-home"
  >
    <div>
      <Bootstrap(undefined)>
        <CarouselItem>
          <img
            alt="First slide"
            className="d-block w-100"
            src={false}
          />
          <CarouselCaption />
        </CarouselItem>
        <CarouselItem>
          <img
            alt="Third slide"
            className="d-block w-100"
            src={false}
          />
          <CarouselCaption />
        </CarouselItem>
        <CarouselItem>
          <img
            alt="Third slide"
            className="d-block w-100"
            src={false}
          />
          <CarouselCaption />
        </CarouselItem>
      </Bootstrap(undefined)>
    </div>
    <h3
      id="punch-line"
    >
      Professional Exciting Positive Friendly Fun
    </h3>
  </div>
  <div
    className="footer-container"
  >
    <div
      className="footerBox"
    >
      <Link
        className="admin-link"
        to="/admin/login/"
      >
        <p
          id="admin-link"
        >
          admin login &gt;
        </p>
      </Link>
    </div>
  </div>
</div>
`;
